src/resty/gcp/init.lua

```diff
@@ -79,14 +79,20 @@ local function build_request(accesstoken, apiDetail, baseUrl, params, requestBod
     if apiDetail.flatPath then
         path_template = apiDetail.flatPath
     elseif mediaUpload then
-        path_template = "https://storage.googleapis.com" .. apiDetail.mediaUpload.protocols.simple.path
+        baseUrl = "https://storage.googleapis.com"
+        path_template = assert(apiDetail.mediaUpload.protocols.simple.path,
+            "we only supported simple path for media upload for now")
     else
-        path_template = baseUrl .. apiDetail.path
+        path_template = apiDetail.path
     end
 
-    local path = template_expansion(path_template, params)
+    local path = baseUrl .. template_expansion(path_template, params)
     local query
 
+    if apiDetail.flatPath then
+        return path, req
+    end
+
```

Comment:
@StarlightIbuki @flrgh This change prevents query parameters from being added to any APIs that have a `flatPath` defined, and most have.  It also prevents parameters from being validated.  I would like to remove this early return, but I'm wondering if we have enough tests to verify that things still work.  Any comments would be appreciated.